velox/type/Filter.cpp

```diff
@@ -276,6 +285,227 @@ bool BigintValuesUsingHashTable::testInt64Range(
   return max >= *it;
 }
 
+NegatedBigintValuesUsingBitmask::NegatedBigintValuesUsingBitmask(
+    int64_t min,
+    int64_t max,
+    const std::vector<int64_t>& values,
+    bool nullAllowed)
+    : Filter(true, nullAllowed, FilterKind::kNegatedBigintValuesUsingBitmask),
+      min_(min),
+      max_(max) {
+  VELOX_CHECK(min <= max, "min must be no greater than max");
+
+  bitmask_.resize(max - min + 1);
+
+  for (int64_t value : values) {
+    bitmask_[value - min] = true;
+  }
+}
+
+bool NegatedBigintValuesUsingBitmask::testInt64(int64_t value) const {
+  if (value < min_ || value > max_) {
+    return true;
+  }
+  return !bitmask_[value - min_];
+}
+
+std::vector<int64_t> NegatedBigintValuesUsingBitmask::values() const {
+  std::vector<int64_t> values; // values that are NOT accepted
+  for (int i = 0; i < bitmask_.size(); i++) {
+    if (bitmask_[i]) {
+      values.push_back(min_ + i);
+    }
+  }
+  return values;
+}
+
+bool NegatedBigintValuesUsingBitmask::testInt64Range(
+    int64_t min,
+    int64_t max,
+    bool hasNull) const {
+  if (hasNull && nullAllowed_) {
+    return true;
+  }
+
+  if (min == max) {
+    return testInt64(min);
+  }
+
+  // boundary checks: if max < min_ then min < min_ as well and vice versa
+  // since part of the range is outside the bitmask, the endpoint is accepted
+  if (min < min_ || max > max_)
+    return true;
+
+  // check the range manually
+  for (int i = min - min_; i <= max - min_; ++i)
+    if (!bitmask_[i])
+      return true;
+
+  return false;
+}
+
+NegatedBigintValuesUsingHashTable::NegatedBigintValuesUsingHashTable(
+    int64_t min,
+    int64_t max,
+    const std::vector<int64_t>& values,
```

Comment:
because this is being passed as a const, copy elision wont take place. Any reason not to pass in an rvalue reference (&&).